python/mrc/_pymrc/src/utilities/object_wrappers.cpp

```diff
@@ -0,0 +1,172 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2021-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "pymrc/utilities/object_wrappers.hpp"
+
+#include "mrc/exceptions/runtime_error.hpp"
+
+namespace mrc::pymrc {
+
+namespace py = pybind11;
+
+PyObjectWrapper::PyObjectWrapper(pybind11::object&& to_wrap) : m_obj(std::move(to_wrap)) {}
+
+PyObjectWrapper::~PyObjectWrapper()
```

Comment:
Do we guarantee that this object goes out of scope before the python interpreter exits?

Reply:
No we dont. This object exists currently in MRC. Would be better to address this in a separate issue